src/modules/job-info/lookup.c

```diff
@@ -288,17 +406,29 @@ static int check_keys_for_eventlog (struct lookup_ctx *l)
         return 0;
     }
 
-    json_array_foreach(l->keys, index, key) {
-        const char *keystr;
-        if (!(keystr = json_string_value (key))) {
-            errno = EINVAL;
-            return -1;
+    return 0;
+}
+
+/* If we need the eventlog for an allow check or for update-lookup
+ * we need to add it to the key lookup list.
+ */
+static int check_to_lookup_eventlog (struct lookup_ctx *l)
+{
+    if (!l->allow || (l->flags & FLUX_JOB_LOOKUP_UPDATE)) {
+        size_t index;
+        json_t *key;
+        json_array_foreach(l->keys, index, key) {
+            const char *keystr;
+            if (!(keystr = json_string_value (key))) {
+                errno = EINVAL;
+                return -1;
+            }
+            if (streq (keystr, "eventlog"))
+                return 0;
         }
```

Comment:
It would probably be good to check that the request `keys` is an array of strings right in the request handler rather than deeper in the code, for clarity and to avoid repeating the same check in several places.  Then this could just be:
```c
if (streq (json_string_value (key), "eventlog"))
    return 0;
```